src/hasher.rs

```diff
@@ -86,29 +86,32 @@ impl Hasher {
   }
 
   fn hash_file_io(&mut self, file: &Path) -> io::Result<(Option<md5::Digest>, Bytes)> {
```

Comment:
Instead of overloading `hash_file_io` to handle both cases, how about creating a `hash_read` function that takes a  `&dyn Read` argument, which is the input stream to hash? I think that would result in two simpler functions, as opposed to a more complex overloaded function?

Reply:
I think that `hash_file_io` can call `hash_read` with the opened file

Reply:
Or maybe `hash_file`would create the the `dyn Read` and pass it to `hash_read`?

Reply:
Right, i guess the innermost function would be `hash_read_io`, which takes a `&mut dyn Read` and returns an `io::Result`